src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -851,6 +852,43 @@ void ShenandoahHeap::handle_promotion_failure() {
   old_heuristics()->handle_promotion_failure();
 }
 
+void ShenandoahHeap::report_promotion_failure(Thread* thread, size_t size) {
+  // We squelch excessive reports to reduce noise in logs.  Squelch enforcement is not "perfect" because
+  // this same code can be in-lined in multiple contexts, and each context will have its own copy of the static
+  // last_report_epoch and this_epoch_report_count variables.
```

Comment:
Is this comment still relevant.  Maybe not since the code is no longer in-lined in multiple contexts.

Thanks for this improvement to the code structure.  This should be relatively rare, so we don't want to in-line it everywhere.